rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -0,0 +1,259 @@
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == amount
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.counterparty = CPT_HOP
+                event.notes = f'Bridge {event.balance.amount} {event.asset.symbol_or_name()} via Hop protocol'  # noqa: E501
+                break
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_events(self, context: DecoderContext) -> DecodingOutput:
+        """This function is used to decode the bridging events done via Hop protocol."""
+        if context.tx_log.topics[0] == WITHDRAWAL_BONDED:
+            self._decode_withdrawal_bonded(context=context)
+
+        if context.tx_log.topics[0] == TRANSFER_SENT:
+            self._decode_transfer_sent(context=context)
+
+        if context.tx_log.topics[0] == WITHDREW:
+            self._decode_withdrew(context=context)
+
+        if context.tx_log.topics[0] == TRANSFER_FROM_L1_COMPLETED:
+            recipient = hex_or_bytes_to_address(context.tx_log.topics[1])
+            if not self.base.is_tracked(recipient):
+                return DEFAULT_DECODING_OUTPUT
+
+            if (bridge := self.bridges.get(context.tx_log.address)) is None:
+                return DEFAULT_DECODING_OUTPUT
+
+            amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+            amount = self._get_amount(amount_raw, bridge)
+
+            for event in context.decoded_events:
+                if event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.NONE and recipient == event.location_label and event.asset == bridge.asset:  # noqa: E501
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.BRIDGE
+                    event.counterparty = CPT_HOP
+                    event.notes = f'Bridge {event.balance.amount} {event.asset.symbol_or_name()} via Hop protocol'   # noqa: E501
+                    event.extra_data = {'sent_amount': str(amount)}
+                    break
+
+        return DEFAULT_DECODING_OUTPUT
+
+    # -- DecoderInterface methods
+    def post_decoding_rules(self) -> dict[str, list[tuple[int, Callable]]]:
+        return {CPT_HOP: [(-1, self._decode_post_events)]}
+
+    def addresses_to_counterparties(self) -> dict[ChecksumEvmAddress, str]:
+        return dict.fromkeys({bridge.address for bridge in self.bridges.values()}, CPT_HOP)
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return dict.fromkeys({bridge.address for bridge in self.bridges.values()}, (self._decode_events,))  # noqa: E501
+
+    @ staticmethod
+    def counterparties() -> tuple[CounterpartyDetails, ...]:
+        return (HOP_CPT_DETAILS,)
```

Comment:
does this even work?

```suggestion
    @staticmethod
    def counterparties() -> tuple[CounterpartyDetails, ...]:
        return (HOP_CPT_DETAILS,)
```

Reply:
yeah for some reason, no lint errors either